src/dashmips.ts

```diff
@@ -25,7 +25,9 @@ export interface DashmipsDebugClient {
 
 export class DashmipsDebugClient extends EventEmitter {
     public dashmipsPid: number = -1
-    public open: boolean
+    public open: boolean = false
+    public verified: boolean = false
+    public stopEntry: boolean = false
```

Comment:
Can we make these Subject's instead? The Subject is for this use case where you want to check that some "subject" (I guess better than saying some "thing") is available, if not you'll wait until some timeout. and then proceed. If the Subject is available then you call .notify() and that will make one of the waiters wake up, (notifyAll to let everyone know). This is how we can potentially not wait the full amount of time which might be a better UX.

Anyway just to be clear, basically where you set `open = true` is where you'd call `open.notifyAll()` and where you check with an if stmt and sleep is where you'll call `client.open.wait()`

I wrapped this functionality in client .ready() but that was for our websocket days and maybe its not best to wrap as there is a need for the specific difference between open and verified etc..

Reply:
Added a new commit, changed open and verified to subjects (I assumed stopEntry should continue to be a bool, as we use that to check if there is a breakpoint and not to wait) and the extension still works as intended!